Overheating UI + Added Overheating to all lasers

oninoni opened the pull request.

dhkatz reviewed (changes requested):
A lot of overhaul has been done internally. Expect to need to change things soon or even cancel PR

oninoni commented:
Is there even overheating atm?

oninoni committed:
Shield Regen + Damage Optimisation

- Shield now Regens after a given time
- If more damage than there is shield left it applies damage to hull.
